core/src/main/java/jenkins/security/QueueItemAuthenticatorMonitor.java

```diff
@@ -93,7 +105,7 @@ public static boolean isQueueItemAuthenticatorConfigured() {
 
     public boolean isAnyBuildLaunchedAsSystemWithAuthenticatorPresent() {
         // you configured a QueueItemAuthenticator, but builds are still running as SYSTEM
-        return anyBuildLaunchedAsSystemWithAuthenticatorPresent;
+        return !buildsLaunchedAsSystemWithAuthenticatorPresentByJob.isEmpty();
```

Comment:
Technically, you should synchronize here for reading the value so you don't get half-written state info. In practice, it might not be an actual issue.

Reply:
Yeah, UI only and the only potential effect I could imagine is a very short term lack of the monitor, so didn't care enough. Happy to add if there's a desire to have it.

Reply:
If you add `@GuardedBy` to the field, findbugs will help identify any missing synchronized blocks.